ICSSoft.STORMNET.Business/SQLDataService/SQLDataService.Async.cs

```diff
@@ -0,0 +1,452 @@
+                {
+                    res = Utils.ProcessingRowsetData(
+                            resValue, dataObjectType, storageStruct, customizationStruct, this, Types, dataObjectCache, SecurityManager, connection, transaction);
+                }
+
+                return res;
+            }
+            finally
+            {
+                dataObjectCache.StopCaching();
+            }
+        }
+
+        /// <inheritdoc cref="IAsyncDataService.LoadObjectsAsync(LoadingCustomizationStruct, DataObjectCache)"/>
+        public virtual Task<DataObject[]> LoadObjectsAsync(LoadingCustomizationStruct customizationStruct, DataObjectCache dataObjectCache = null)
+        {
+            if (dataObjectCache == null)
+            {
+                dataObjectCache = new DataObjectCache();
+            }
+
+            RunChangeCustomizationString(customizationStruct.LoadingTypes);
+            return LoadObjectsByExtConnAsync(customizationStruct, dataObjectCache, GetDbConnection());
+        }
+
+        /// <inheritdoc cref="IAsyncDataService.LoadObjectsAsync(View, DataObjectCache)"/>
+        public virtual Task<DataObject[]> LoadObjectsAsync(View dataObjectView, DataObjectCache dataObjectCache = null)
+        {
+            if (dataObjectCache == null)
+            {
+                dataObjectCache = new DataObjectCache();
+            }
+
+            LoadingCustomizationStruct lc = new LoadingCustomizationStruct(GetInstanceId());
+            lc.View = dataObjectView;
+            lc.LoadingTypes = new[] { dataObjectView.DefineClassType };
+            return LoadObjectsAsync(lc, dataObjectCache);
+        }
+
+        /// <summary>
+        /// Асинхронная вычитка данных.
+        /// </summary>
+        /// <param name="query">Запрос для вычитки.</param>
+        /// <param name="loadingBufferSize">Ограничение на количество строк, которые будут загружены.</param>
+        /// <returns>Асинхронная операция (возвращает результат вычитки).</returns>
+        public virtual Task<object[][]> ReadAsync(string query, int loadingBufferSize)
+        {
+            return ReadAsyncByExtConn(query, loadingBufferSize, GetDbConnection());
+        }
+
+        /// <summary>
+        /// Асинхронная вычитка данных.
+        /// </summary>
+        /// <param name="query">Запрос для вычитки.</param>
+        /// <param name="loadingBufferSize">Количество строк, которые нужно загрузить в рамках текущей вычитки (используется для повторной дочитки).</param>
+        /// <param name="connection">Соединение, в рамках которого нужно выполнить запрос (если соединение закрыто - оно откроется).</param>
+        /// <param name="transaction">Транзакция, в рамках которой выполняется запрос.</param>
+        /// <returns>Асинхронная операция (возвращает результат вычитки).</returns>
+        public virtual async Task<object[][]> ReadAsyncByExtConn(string query, int loadingBufferSize, DbConnection connection, DbTransaction transaction = null)
```

Comment:
```suggestion
        public virtual async Task<object[][]> ReadByExtConnAsync(string query, int loadingBufferSize, DbConnection connection, DbTransaction transaction = null)
```
Лучше ключевое слово `Async` в конец названия метода

Reply:
Кроме того аналогично #64 нужно использовать wrapper

Reply:
Нужен ли он здесь? В оригинальном методе также (через connection + transaction).

Reply:
Всё-таки сделал через wrapper - нашёл место где коннекция раньше открывается.